doc/signal_path.rst

```diff
@@ -0,0 +1,302 @@
+   \draw[->, dotted] (delays) to[lbl, auto, edge label=f32] (eqx);
+   \draw[->, dotted] (delays) to[lbl, auto, edge label=i32] (cdelayy);
+   \draw[->, dotted] (delays) to[lbl, auto, edge label'=f32] (eqy);
+   \draw[->, dotted] (eq) to[lbl, auto, edge label'=cf32] (eqx);
+   \draw[->, dotted] (eq) to[lbl, auto, edge label=cf32] (eqy);
+   \draw[->] (quantx) |- node[lbl, auto, swap, near start] {ci8} (pack);
+   \draw[->] (quanty) |- node[lbl, auto, near start] {ci8} (pack);
+
+The mixer frequency is quantised to a 0.32 unsigned fixed-point representation
+of the number of cycles per post-DDC sample. The resolution
+is thus :math:`2^{-31}B` where :math:`B` is the final bandwidth.
+
+Discarding half the channels after channelisation allows for a lot of freedom
+in the design of the DDC FIR filter: the discarded channels can have an
+arbitrary response. This allows for a gradual transition from passband to
+stopband. We use :func:`scipy.signal.remez` to produce a filter that is as
+close as possible to 1 in the passband and 0 in the stopband. A weighting
+factor (which the user can override) balances the priority of the passband
+(ripple) and stopband (alias suppression).
+
+The filter performance is slightly improved by noting that the discarded
+channels have multiple aliases, and the filter response in those aliases is
+also irrelevant. We thus use :func:`scipy.signal.remez` to only optimise the
+response to those channels that alias into the output.
+
+Correlation
+-----------
+Given a baseline (p, q) and time-varying channelised voltages :math:`e_p` and
+:math:`e_q`, the correlation product is the sum of :math:`e_p \overline{e_q}`
+over the accumulation period. This is computed in integer arithmetic and so is
+lossless except when saturation occurs.
+
+The figure below shows the signal path.
+
+.. tikz:: Signal path for correlation
+   :libs: chains
+
+   \tikzset{
+     base/.style={minimum width=2.5cm, minimum height=1cm, align=center},
+     op/.style={draw, base},
+     control/.style={draw, base, rounded corners, dotted},
+     lbl/.style={font=\scriptsize},
+     every join/.style={draw,->},
+     >=latex,
+   }
+   \begin{scope}[start chain=going right]
+     \node[op, on chain] {Receive};
+     \node[op, on chain, join=by {lbl,edge label=ci8}] {Correlate\\ Accumulate};
+     \node[op, on chain, join=by {lbl,edge label=ci64}] {Saturate};
+     \node[op, on chain, join=by {lbl,edge label=ci32}] {Transmit};
+   \end{scope}
+
+Beamforming
+-----------
+The signal path below is repeated for each single-polarisation beam. Delays
+are computed purely with a phase slope in the frequency domain, similarly to
+the fine delays in the channeliser. Dithering is done the same way as for
+channelisation. Since all calculations are performed in single precision
+floating point and the input has a limited range, overflow can only occur
+during quantisation (which saturates).
+
+.. tikz:: Signal path for beamforming
```

Comment:
Observation: the landscape orientation of this diagram make it a bit harder to read compared with the F-engine ones.

The simple X-engine one was easier but this one has more packed into it.